connectors/sources/jira.py

```diff
@@ -320,8 +371,310 @@ def get_default_configuration(cls):
+                "source": {
+                    "bool": {
+                        "filter": {
+                            "bool": {
+                                "should": [
+                                    {
+                                        "bool": {
+                                            "must_not": {
+                                                "exists": {"field": ACCESS_CONTROL}
+                                            }
+                                        }
+                                    },
+                                    {
+                                        "terms": {
+                                            f"{ACCESS_CONTROL}.enum": filtered_access_control
+                                        }
+                                    },
+                                ]
+                            }
+                        }
+                    }
+                },
+            }
+        }
+
+    def _is_active_atlassian_user(self, user_info):
+        user_url = user_info.get("self")
+        if not user_url:
+            self._logger.debug(
+                f"Skipping user: {user_info.get('displayName', 'user')} due to the absence of a personal URL."
+            )
+            return False
+
+        if not user_info.get("active"):
+            self._logger.debug(
+                f"Skipping user: {user_info.get('displayName', 'user')} as it is inactive or deleted."
+            )
+            return False
+
+        if user_info.get("accountType") != ATLASSIAN:
+            self._logger.debug(
+                f"Skipping user: {user_info.get('displayName', 'user')} because the account type is {user_info.get('accountType')}. Only 'atlassian' account type is supported."
+            )
+            return False
+
+        return True
+
+    def _decorate_with_access_control(self, document, access_control):
+        if self._dls_enabled():
+            document[ACCESS_CONTROL] = list(
+                set(document.get(ACCESS_CONTROL, []) + access_control)
+            )
+
+        return document
+
+    async def _user_information_list(self, project):
+        async for response in self.jira_client.api_call(
+            url_name=PROJECT_PERMISSIONS, project_key=project["key"]
+        ):
+            yield await response.json()
+
+    async def _project_access_control(self, project):
+        if not self._dls_enabled():
+            return []
+
+        access_control = set()
+        async for actors in self._user_information_list(project=project):
+            for actor in actors:
+                if actor["accountType"] == ATLASSIAN:
+                    access_control.add(
+                        _prefix_account_id(user_id=actor.get("accountId"))
+                    )
+                    access_control.add(_prefix_username(user=actor.get("displayName")))
+        return list(access_control)
+
+    async def _issue_security_level(self, issue_key):
```

Comment:
Same as above: can probably live inside the `JiraClient`.

Reply:
https://github.com/elastic/connectors-python/issues/1417